packages/amplify-provider-awscloudformation/src/cloud-formation-error-handler.ts

```diff
@@ -0,0 +1,46 @@
+import { $TSAny, AmplifyError } from 'amplify-cli-core';
+import { deserializeErrorMessages, CFNErrorMessage, CFNErrorMessages } from './aws-utils/cloudformation-error-serializer';
+
+const s3Indicator = '(AWS::S3::Bucket)';
+
+const handleS3Error = (err: Error & { details?: string }): void => {
+  const alreadyExistsSuffix = 'already exists';
+  const deserializedErrorMessages: CFNErrorMessages = deserializeErrorMessages(err.details);
+  const bucketExistsLines: Array<CFNErrorMessage> = deserializedErrorMessages.messages.filter(
+    (message) =>
+      message.name.includes(s3Indicator) && message.reason.includes(alreadyExistsSuffix) && !message.name.includes('DeploymentBucket'),
```

Comment:
nit: Similar to below a regex can be used instead of multiple includes
```ts
const bucketError = 'Name: S3Bucket (AWS::S3::Bucket), Event Type: create, Reason: some-bucket-dev already exists\n'
const bucketRegex = /Name: (S3Bucket|CustomMessageConfirmationBucket) \(AWS::S3::Bucket\), .* already exists.*/;
console.log(bucketRegex.test(bucketError));
```